apps/prairielearn/elements/pl-matching/pl-matching.py

```diff
@@ -293,8 +294,6 @@ def parse(element_html: str, data: pl.QuestionData) -> None:
             data["format_errors"][expected_html_name] = (
                 "The submitted answer was left blank."
             )
-        elif student_answer is None:
-            data["format_errors"][expected_html_name] = "No answer was submitted."
```

Comment:
While I agree that we'll never hit this case, I do wonder if we should separately handle the case where no answer was submitted. Right now we rely on `int(None)` raising a `TypeError`, but IMO a separate explicit check for that could make more sense. We do this in many other elements:

https://github.com/PrairieLearn/PrairieLearn/blob/610a427bb6425f1f20c25542f66a33a13eb6310e/apps/prairielearn/elements/pl-multiple-choice/pl-multiple-choice.py#L624-L630

Reply:
I think it's useful as a sanity check, and for consistency with other elements. "option wasn't selected" is different from "form input wasn't present", and other elements handle those separately too, where appropriate. `pl-integer-input` has a better example of this:

https://github.com/PrairieLearn/PrairieLearn/blob/a768f5dd12cd2c4b9936eb0c7384a55cf0cc8089/apps/prairielearn/elements/pl-integer-input/pl-integer-input.py#L224-L230

Reply:
Sounds good.

Reply:
Resolved in 87d8d3e1e5023ebc1a4a6af1edda3dfccfdc41ff.

Reply:
LGTM. Someday I'll get used to try: ... except KeyError:` as the way of doing that lmao. Still reads so much worse to me than linear checks.